### PR TITLE
Code review suggestions for apache/accumulo#1873

### PR DESCRIPTION
**Pull request by ctubbsii**
These suggestions do the following:

* Remove Guava annotation (this ties us to unneeded dependency on Guava,
  and it adds very little value that a simple inline comment can
  provide, especially since it's not even public API)
* Remove non-standard warnings suppression for a warning type that does
  not exist in the JDK
* Simplify the PropertyRenamer interface to be a simple class POJO for
  carrying a pair of a matcher predicate and a transforming function
  (this reduces the number of lines in the PR significantly)
* Remove boolean argument to overloaded renameDeprecatedProperty method
  and create a separate method for the occurrences when a warning is
  desired
* Remove unneeded deprecation annotations for internal classes/methods
  that aren't user-facing. This simplifies warning suppression/handling,
  by eliminating propagation warnings in the migration code and its
  tests
* Reuse the predicate as much as possible, to avoid re-implementing the
  matcher "startsWith" checks in different places
* Add missing override annotations in IT

### CODE REVIEW

File: core/src/main/java/org/apache/accumulo/core/conf/Property.java [hunk 1]
```diff
@@ -1075,7 +1075,7 @@
   private Property replacedBy = null;
   private PropertyType type;
 
-  private Property(String name, String defaultValue, PropertyType type, String description) {
+  Property(String name, String defaultValue, PropertyType type, String description) {
```
Comment (brianloss) [inline core/src/main/java/org/apache/accumulo/core/conf/Property.java:1078]: This change wasn't in my original PR. I'm just curious why access to this constructor is expanded here?

**Comment by brianloss** [comment]
> * Remove Guava annotation (this ties us to unneeded dependency on Guava,
>   and it adds very little value that a simple inline comment can
>   provide, especially since it's not even public API)

I found 47 other uses of `@VisibleForTesting` in the code base. Should we open a ticket to remove those?

> * Remove non-standard warnings suppression for a warning type that does
>   not exist in the JDK

Thank IntelliJ for this. I have seen it insert special comments when you suppress inspections, but I didn't realize it would insert `@SuppressWarnings` with tags that aren't JDK supported.

> * Remove unneeded deprecation annotations for internal classes/methods
>   that aren't user-facing. This simplifies warning suppression/handling,
>   by eliminating propagation warnings in the migration code and its
>   tests

My thinking was this would be an easy way to identify code that can be removed in the next major release. If nothing at all is deprecated, then aren't we more likely to forget by the time the 3.0 release rolls around?

File: core/src/main/java/org/apache/accumulo/core/conf/Property.java (hunk 1 above)
Comment (ctubbsii) [reply core/src/main/java/org/apache/accumulo/core/conf/Property.java:1078]: It's not expanded. It's an enum. My IDE automatically removed extraneous keywords. It's a quirk of Java that enum constructors are always private, whether the `private` keyword is present or not (`public` and `protected` keywords cause a compile error).

**Comment by ctubbsii** [comment]
> > * Remove Guava annotation (this ties us to unneeded dependency on Guava,
> >   and it adds very little value that a simple inline comment can
> >   provide, especially since it's not even public API)
> 
> I found 47 other uses of `@VisibleForTesting` in the code base. Should we open a ticket to remove those?

Maybe. It's not that important. I've mentioned it in code reviews, to avoid additional reliance on Guava, but have not taken the time to remove existing uses. If we do that, we could add a checkstyle rule to ban its use in future... but we'd have to evaluate each current use to see if they are necessary or not.

We have a dependency on Guava either way. But, new uses are easier to question and avoid. The less we use it when we don't need to, the easier it will be to clean up our dependencies that aren't actually needed later.

> 
> > * Remove non-standard warnings suppression for a warning type that does
> >   not exist in the JDK
> 
> Thank IntelliJ for this. I have seen it insert special comments when you suppress inspections, but I didn't realize it would insert `@SuppressWarnings` with tags that aren't JDK supported.
> 

Yeah, Eclipse has its own warnings also. I figure the fewer warnings we generate, the fewer we need to suppress, as a general rule. When we do need to suppress them, it'd be nice to stick to JDK-supported ones, so we're not tied to specific IDEs.

> > * Remove unneeded deprecation annotations for internal classes/methods
> >   that aren't user-facing. This simplifies warning suppression/handling,
> >   by eliminating propagation warnings in the migration code and its
> >   tests
> 
> My thinking was this would be an easy way to identify code that can be removed in the next major release. If nothing at all is deprecated, then aren't we more likely to forget by the time the 3.0 release rolls around?

For that, I was thinking we leave a `MASTER_PREFIX` enum in `Property.java`, as deprecated. The other thing is that this exists entirely in a `DeprecationUtils` class, so it should be easy to review. We can also preemptively create a blocker issue for 3.0 to remind us to take care of it. There are also problems using `forRemoval = true`, because they propagate differently than `forRemoval = false`, and some IDEs don't understand the new rules (see https://bugs.eclipse.org/bugs/show_bug.cgi?id=565271)

**Pull request merged**